### PR TITLE
Pub 2254 - Upload Publication Test

### PR DESCRIPTION
**Pull request by ashwini-mv**
### Jira link (if applicable)

https://tools.hmcts.net/jira/browse/PUB-2253

### Change description ###

Performance Testing - Upload Publication Test

### Checklist

- [x] commit messages are meaningful and follow good commit message guidelines
- [x] README and other documentation has been updated / added (if needed)
- [x] tests have been updated / new tests has been added (if needed)
- [x] Does this PR introduce a breaking change

### CODE REVIEW

File: src/gatling/scala/config/ApiConfig.scala [hunk 1]
```diff
@@ -5,7 +5,7 @@ import com.fasterxml.jackson.annotation.JsonProperty
 class ApiConfig(@JsonProperty("scopes") _scope: String,
                 @JsonProperty("url") _url: String
                ) {
-  val scope = _scope
-  val url = _url
+ lazy val scope = _scope
```
Comment (ChrisS1512) [inline src/gatling/scala/config/ApiConfig.scala:8]: Is this still needed to be marked as 'Lazy'?

File: src/gatling/scala/config/RootConfig.scala [hunk 2]
```diff
@@ -19,4 +19,9 @@ class RootConfig(@JsonProperty("tenantId") _tenantId: String,
   val accountManagementApi = _accountManagementApi
   val channelManagementApi = _channelManagementApi
   val publicationServicesApi = _publicationServicesApi
+  val rampUpUsers: Int = Integer.getInteger("rampUsers", 0).toInt
```
Comment (ChrisS1512) [inline src/gatling/scala/config/RootConfig.scala:22]: Not sure these are needed. Could you try to remove?

File: src/gatling/scala/simulations/publication/CreatePublicationCivilAndFamilySimulation.scala [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+class CreatePublicationCivilAndFamilySimulation extends Simulation {
+
+  print("api url is" + config.dataManagementApi.url)
+  private val createPublicationCivilFamilyExec = PublicationScenarios.CreatePublicationCivilAndFamilyScenario
+    .inject(
+      atOnceUsers(onceUsers),
+      rampUsers(rampUpUsers) during(rampUpUsersDuration seconds)
+    )
+  // Per-06
```
Comment (ChrisS1512) [inline src/gatling/scala/simulations/publication/CreatePublicationCivilAndFamilySimulation.scala:20]: The // Per-06 comments I think can be removed. Similar in the other files below

File: src/gatling/scala/simulations/publication/CreatePublicationCivilAndFamilySimulation.scala [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+import scenarios.publication.PublicationScenarios
+import utils.auth.OAuthAPI.config
+import utils.auth.OAuthAPI.config.{onceUsers, rampUpUsers, rampUpUsersDuration}
+
+import scala.concurrent.duration.DurationInt
+import scala.language.postfixOps
+
+class CreatePublicationCivilAndFamilySimulation extends Simulation {
+
+  print("api url is" + config.dataManagementApi.url)
```
Comment (ChrisS1512) [inline src/gatling/scala/simulations/publication/CreatePublicationCivilAndFamilySimulation.scala:14]: This print statement can be removed

**Review by ChrisS1512 — CHANGES_REQUESTED** [review]

File: src/gatling/scala/simulations/publication/CreatePublicationCivilAndFamilySimulation.scala (hunk 4 above)
Comment (ashwini-mv) [reply src/gatling/scala/simulations/publication/CreatePublicationCivilAndFamilySimulation.scala:14]: Done :)

File: src/gatling/scala/simulations/publication/CreatePublicationCivilAndFamilySimulation.scala (hunk 3 above)
Comment (ashwini-mv) [reply src/gatling/scala/simulations/publication/CreatePublicationCivilAndFamilySimulation.scala:20]: Done :)

File: src/gatling/scala/config/RootConfig.scala (hunk 2 above)
Comment (ashwini-mv) [reply src/gatling/scala/config/RootConfig.scala:22]: We are using this in simulations

File: src/gatling/scala/config/ApiConfig.scala (hunk 1 above)
Comment (ashwini-mv) [reply src/gatling/scala/config/ApiConfig.scala:8]: Removed :)

**Review by ChrisS1512 — APPROVED** [review]

**Pull request merged**